### PR TITLE
Fix link generation for ABC iview

### PR DESCRIPTION
**Pull request by ryleyangus**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/rg3/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/rg3/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/rg3/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Bug fix
- [ ] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

The ABC recently changed their streaming platform to block non-official clients. This pull request reflects the changes made to the iOS app to generate valid HLS links. As I do not have the key required for the HDS links, I also had to change the higher quality source URL to the HLS option. This fixes Issue #14711

### CODE REVIEW

File: youtube_dl/extractor/abc.py [hunk 1]
```diff
@@ -126,12 +129,19 @@ def _real_extract(self, url):
         title = video_params.get('title') or video_params['seriesTitle']
         stream = next(s for s in video_params['playlist'] if s.get('type') == 'program')
 
-        format_urls = [
-            try_get(stream, lambda x: x['hds-unmetered'], compat_str)]
+        TIME = '%.6f' % time.time()
+        URLPARAM = '/auth/hls/sign?ts={}&hn={}&d=(null)'.format(TIME, video_id)
```
Comment (dstftw) [inline youtube_dl/extractor/abc.py]: No `{}`.

File: youtube_dl/extractor/abc.py [hunk 2]
```diff
@@ -126,12 +129,19 @@ def _real_extract(self, url):
...
+        message = str(URLPARAM).encode('utf-8')
```
Comment (dstftw) [inline youtube_dl/extractor/abc.py]: str is pointless here.

File: youtube_dl/extractor/abc.py [hunk 3]
```diff
@@ -126,12 +129,19 @@ def _real_extract(self, url):
...
+        secret = str('MainContainerViewController').encode('utf-8')
```
Comment (dstftw) [inline youtube_dl/extractor/abc.py]: Same.

File: youtube_dl/extractor/abc.py [hunk 4]
```diff
@@ -126,12 +129,19 @@ def _real_extract(self, url):
...
+        SIG = hmac.new(secret, message, digestmod=hashlib.sha256).hexdigest()
+        abc_signature_url = "http://iview.abc.net.au" + URLPARAM + "&sig=" + SIG
```
Comment (dstftw) [inline youtube_dl/extractor/abc.py]: Single quotes.

File: youtube_dl/extractor/abc.py [hunk 5]
```diff
@@ -126,12 +129,19 @@ def _real_extract(self, url):
...
+        urlsuffix = "?hdnea=" + self._download_webpage(abc_signature_url, None)
```
Comment (dstftw) [inline youtube_dl/extractor/abc.py]: `None` is not a video id.

**Comment by ryleyangus** [comment]
Thanks for the feedback dstftw. I wasn't entirely too sure about the preferred formatted string method. Hopefully it is better now.

**Comment by zoocey** [comment]
There is also https://github.com/rg3/youtube-dl/pull/14782 which fixes the same issue.

**Pull request closed without merging**